cmd/jaeger/internal/unittest/unittest.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package unittest
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.opentelemetry.io/collector/service/telemetry"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/exporters/storageexporter"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+type storageHost struct {
+	t                *testing.T
+	storageExtension component.Component
+}
+
+type StorageTest struct {
+	Name       string
+	ConfigFile string
+}
+
+func (s *StorageTest) Test(t *testing.T) {
+	if os.Getenv("STORAGE") != s.Name {
+		t.Skipf("Integration test against Jaeger-V2 %[1]s skipped; set STORAGE env var to %[1]s to run this", s.Name)
+	}
+	var v integration.StorageIntegration
+	factories, err := internal.Components()
```

Comment:
@james-ryans if feels like there is a lot of repetition here compared to `cmd/jaeger/internal/integration/integration.go`.